app/credentials/smtp_credentials.rb

```diff
@@ -1,17 +1,37 @@
+require 'concerns/user_provided_service'
+
 class SMTPCredentials
+  extend UserProvidedService
+
   def self.smtp_password
-    ENV["MICROPURCHASE_SMTP_SMTP_PASSWORD"]
+    if use_env_var?
+      ENV['MICROPURCHASE_SMTP_SMTP_PASSWORD']
+    else
+      credentials('micropurchase-smtp')['smtp_password']
+    end
   end
 
   def self.smtp_username
-    ENV["MICROPURCHASE_SMTP_SMTP_USERNAME"]
+    if use_env_var?
+      ENV['MICROPURCHASE_SMTP_SMTP_USERNAME']
+    else
+      credentials('micropurchase-smtp')['smtp_username']
+    end
   end
 
   def self.default_url_host
-    ENV["MICROPURCHASE_SMTP_DEFAULT_URL_HOST"]
+    if use_env_var?
+      ENV['MICROPURCHASE_SMTP_DEFAULT_URL_HOST']
+    else
+      credentials('micropurchase-smtp')['default_urL_host']
```

Comment:
that's how we have our user-provided services now -- all lowercase. service names are dasherized, keys are underscore